vm/src/dictdatatype.rs

```diff
@@ -223,6 +240,9 @@ impl<T: Clone> Dict<T> {
                 let mut inner = self.write();
                 // Update existing key
                 if let Some(entry) = inner.entries.get_mut(index) {
+                    let entry = entry
+                        .as_mut()
+                        .expect("The dict was changed since we did lookup. Let's try again.");
```

Comment:
`expect` is a sort of `panic`. So it will not try again anything. Just `"The dict was changed since we did lookup."` will be enough